### PR TITLE
Support Bytes type for operands of RelOpValue (EQ, NEQ, NULL, NOT_NULL)

### PR DESCRIPTION
**Pull request by arnaud-lacurie**

### CODE REVIEW

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/LiteralValue.java [hunk 1]
```diff
@@ -107,7 +107,10 @@ public boolean equalsWithoutChildren(@Nonnull final Value other, @Nonnull final
         }
 
         final LiteralValue<?> that = (LiteralValue<?>)other;
-        return Objects.equals(value, that.value);
+        if (value == null && that.value == null) {
```
Comment (normen662) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/LiteralValue.java:110]: Nice! Trap avoided!

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java [hunk 2]
```diff
@@ -734,7 +743,10 @@ private enum UnaryPhysicalOperator {
         IS_NOT_NULL_FI(Comparisons.Type.NOT_NULL, Type.TypeCode.FLOAT, Objects::nonNull),
         IS_NOT_NULL_DI(Comparisons.Type.NOT_NULL, Type.TypeCode.DOUBLE, Objects::nonNull),
         IS_NOT_NULL_SS(Comparisons.Type.NOT_NULL, Type.TypeCode.STRING, Objects::nonNull),
-        IS_NOT_NULL_BI(Comparisons.Type.NOT_NULL, Type.TypeCode.BOOLEAN, Objects::nonNull);
+        IS_NOT_NULL_BI(Comparisons.Type.NOT_NULL, Type.TypeCode.BOOLEAN, Objects::nonNull),
```
Comment (normen662) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java:746]: Not a comment to you but I have not clue what the last prefix represents shouldn't this always be just representing the input type like in your case? I mean what does the `I` in `IS_NOT_NULL_FI` stand for?

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java [hunk 3]
```diff
@@ -648,7 +649,15 @@ private enum BinaryPhysicalOperator {
         GTE_US(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.STRING, (l, r) -> null),
         GTE_UV(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.VERSION, (l, r) -> null),
         GTE_VU(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.VERSION, Type.TypeCode.UNKNOWN, (l, r) -> null),
-        GTE_VV(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.VERSION, Type.TypeCode.VERSION, (l, r) -> ((FDBRecordVersion)l).compareTo((FDBRecordVersion) r) >= 0),;
+        GTE_VV(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.VERSION, Type.TypeCode.VERSION, (l, r) -> ((FDBRecordVersion)l).compareTo((FDBRecordVersion) r) >= 0),
```
Comment (normen662) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java:652]: So we explicitly disallow `ByteString` bytes here?

**Review by normen662 — CHANGES_REQUESTED** [review]

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java (hunk 2 above)
Comment (arnaud-lacurie) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java:746]: I have no clue either. I also don't know why all the other ones have `I` as a suffix, but strings have `S`.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java (hunk 3 above)
Comment (arnaud-lacurie) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java:652]: I guess so, although it's more that tests were passing well with this, so I would not introduce something else unless we think it's necessary.

**Review by normen662 — APPROVED** [review]

**Pull request merged**